File: Haskell/beginner/1101.hs
-- https://judge.beecrowd.com/pt/problems/view/1101
-- Sequência de Números e Soma

-- Imports
import Control.Monad (unless)

-- Main function
main :: IO ()
main = loop

-- loop function
loop :: IO ()
loop = do
    -- Taking values
    input <- getLine
    let [usr_input1, usr_input2] = map read (words input) :: [Int]
    
    -- Stop condition
    unless (usr_input1 <= 0 || usr_input2 <= 0) $ do
        let (minVal, maxVal) = if usr_input1 > usr_input2
                               then (usr_input2, usr_input1)
                               else (usr_input1, usr_input2)
            seqNums = [minVal..maxVal]
            sumSeq = sum seqNums
        
        -- Printing numbers an sum,
        putStrLn $ unwords (map show seqNums) ++ " Sum=" ++ show sumSeq
        
        -- Looping again
        loop

-- By Fernando Serra
-- https://github.com/fernandosserra

File: C/beginner/1101.c
// https://judge.beecrowd.com/pt/problems/view/1101
// Sequência de Números e Soma

// Libraries
#include <stdio.h>

// Main Function
int main(){
    // Starting Code
    int flag = 0;
    int usr_input1, usr_input2;
    int temp_storage, sum;

    // Iteration
    while (flag == 0) {
        scanf("%d %d", &usr_input1, &usr_input2);
        if (usr_input1 <= 0 || usr_input2 <= 0) {
            flag = 1;
            break;
        }
        
        if (usr_input1 > usr_input2) {
            temp_storage = usr_input1;
            usr_input1 = usr_input2;
            usr_input2 = temp_storage;
        }

        for (int i = usr_input1; i <= usr_input2; i++) {
            printf("%d ", i);
            sum += i;
        }

        printf("Sum=%d\n", sum);
        sum = 0;
    }


    // Ending Code
    return 0;
}

// By Fernando Serra
// https://github.com/fernandosserra